dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/CrudControllerAdvice.java

```diff
@@ -187,6 +188,27 @@ public WebMessage illegalQueryExceptionHandler( IllegalQueryException ex )
         return conflict( ex.getMessage(), ex.getErrorCode() );
     }
 
+    @ExceptionHandler( MethodArgumentTypeMismatchException.class )
+    @ResponseBody
+    public WebMessage methodArgumentTypeMismatchException( MethodArgumentTypeMismatchException ex )
+    {
+        if ( ex.getRequiredType() == null )
+        {
+            return defaultExceptionHandler( ex );
```

Comment:
I think we should always make this a bad request. Any type conversion issue should mean the input was invalid. We can just go with a generic message in case we don't have a specific handling for the type.